src/manager/manager.py

```diff
@@ -33,6 +33,7 @@ def reset(self):
 def manager_event_loop(manager: Manager, vehicles: list[Vehicle], time: float):
     _update_manager_vehicle_list(manager, vehicles)
     _compute_and_send_acceleration_commands(manager, vehicles)
+    _remove_vehicle_(manager, vehicles)
```

Comment:
I think  _update_manager_vehicle_list may be the best place to call this function

Reply:
on line 40-42, there is a condition that the vehicle is already in the manager's list. You can kill two birds with one stone, and not have to define another loop